uint32 task
uint32 current_checkpoint
time[] checkpoints_completion
time start_time
time elapsed_time
bool timed_out
bool finished
